sqlmesh/core/engine_adapter/base.py

```diff
@@ -1217,6 +1218,18 @@ def drop_view(
             **kwargs,
         )
 
+    def create_catalog(self, catalog_name: str | exp.Identifier) -> None:
+        return self._create_catalog(exp.parse_identifier(catalog_name, dialect=self.dialect))
+
+    def _create_catalog(self, catalog_name: exp.Identifier) -> None:
+        raise NotImplementedError()
```

Comment:
We need a more reasonable error here indicating what happened and for which engine